src/scripts/ic8/hiv/hiv_filehandlers.py

```diff
@@ -728,33 +728,27 @@ def _turn_workbook_into_df(self, file: Path) -> pd.DataFrame:
             # Extract Step4 values for 2022
             step4_2022 = csv_df[(csv_df['scenario_descriptor'] == 'Step4') & (csv_df['year'] == 2022)]
 
-            # Merge Step4 values with Step13 rows
-            df_step13_updated = csv_df[mask_step13_2022].merge(
-                step4_2022[['country'] + column_names],
-                on='country',
-                suffixes=('', '_step4')
+            # Create a mapping for Step4 values (x, y, z) by country
+            step4_mapping = step4_2022.set_index('country')[column_names]
+
+            # Update all scenarios in 2022 with values from Step4
+            mask_year_2022 = (csv_df['year'] == 2022)
+            csv_df.loc[mask_year_2022, column_names] = csv_df.loc[mask_year_2022].apply(
+                lambda row: step4_mapping.loc[row['country']] if row['country'] in step4_mapping.index else row[
+                    column_names],
+                axis=1
             )
 
-            # Find the original indices for rows in df_step13_updated
-            indices_to_update = csv_df[
-                (csv_df['country'].isin(df_step13_updated['country'])) &
-                (csv_df['scenario_descriptor'] == 'Step13') &
-                (csv_df['year'] == 2022)
-                ].index
-
-            # Replace values in the original DataFrame at these indices
-            csv_df.loc[indices_to_update, column_names] = df_step13_updated[[col + '_step4' for col in column_names]].values
```

Comment:
For all the manipulations going on here (which I understand are to correct some specific issues with the version of the model results being used), I suggest adding some comments to remind us.